pennylane/circuit_graph.py

```diff
@@ -145,11 +146,17 @@ def __init__(self, ops, obs, wires, par_info=None, trainable_params=None):
         for k, op in enumerate(queue):
             op.queue_idx = k  # store the queue index in the Operator
 
-            if hasattr(op, "return_type") and op.return_type is qml.operation.State:
-                # State measurements contain no wires by default, but wires are
-                # required for the circuit drawer, so we recreate the state
-                # measurement with all wires
-                op = qml.measure.MeasurementProcess(qml.operation.State, wires=wires)
+            if hasattr(op, "return_type"):
+                if op.return_type is qml.operation.State:
+                    # State measurements contain no wires by default, but wires are
+                    # required for the circuit drawer, so we recreate the state
+                    # measurement with all wires
+                    op = qml.measure.MeasurementProcess(qml.operation.State, wires=wires)
+
+                elif op.return_type is qml.operation.Sample and op.wires == Wires([]):
+                    # Sampling without specifying wires is treated as sampling all wires
+                    op = qml.measure.MeasurementProcess(qml.operation.Sample, wires=wires)
```

Comment:
Thanks @Jaybsoni for all your work so far, this is very close to being merged in!

However, this line looks like it is not being tested, which is blocking the CI checks